nixos/modules/services/web-apps/openvscode-server.nix

```diff
@@ -0,0 +1,211 @@
+        default = null;
+        description = lib.mdDoc ''
+          Specifies the directory that user data is kept in. Can be used to open multiple distinct instances of Code.
+        '';
+        type = types.nullOr types.str;
+      };
+
+      server-data-dir = mkOption {
+        default = null;
+        description = lib.mdDoc ''
+          Specifies the directory that server data is kept in.
+        '';
+        type = types.nullOr types.str;
+      };
+
+      extensions-dir = mkOption {
+        default = null;
+        description = lib.mdDoc ''
+          Set the root path for extensions.
+        '';
+        type = types.nullOr types.str;
+      };
+
+      telemetry-level = mkOption {
+        default = "off";
+        example = "crash";
+        description = lib.mdDoc ''
+          Sets the initial telemetry level. Valid levels are: 'off', 'crash', 'error' and 'all'.
+        '';
+        type = types.str;
+      };
+
+      connection-token = mkOption {
+        default = null;
+        example = "secret-token";
+        description = lib.mdDoc ''
+          A secret that must be included with all requests.
+        '';
+        type = types.nullOr types.str;
+      };
+
+      connection-token-file = mkOption {
+        default = null;
+        description = lib.mdDoc ''
+          Path to a file that contains the connection token.
+        '';
+        type = types.nullOr types.str;
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.openvscode-server = {
+      description = "OpenVSCode server";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+      path = cfg.extraPackages;
+      environment = cfg.extraEnvironment;
+      serviceConfig = {
+        ExecStart = ''
+          ${cfg.package}/bin/openvscode-server \
+            --accept-server-license-terms \
+            --host=${cfg.host} \
+            --port=${toString cfg.port} \
+          '' + optionalString (cfg.telemetry-level == true) ''
+            --telemetry-level=${cfg.telemetry-level} \
+          '' + optionalString (cfg.without-connection-token == true) ''
+            --without-connection-token \
+          '' + optionalString (cfg.socket-path != null) ''
+            --socket-path=${cfg.socket-path} \
+          '' + optionalString (cfg.user-data-dir != null) ''
+            --user-data-dir=${cfg.user-data-dir} \
+          '' + optionalString (cfg.server-data-dir != null) ''
+            --server-data-dir=${cfg.server-data-dir} \
+          '' + optionalString (cfg.extensions-dir != null) ''
+            --extensions-dir=${cfg.extensions-dir} \
+          '' + optionalString (cfg.connection-token != null) ''
+            --connection-token=${cfg.connection-token} \
+          '' + optionalString (cfg.connection-token-file != null) ''
+            --connection-token-file=${cfg.connection-token-file} \
+          '' + lib.escapeShellArgs cfg.extraArguments;
```

Comment:
Do we need all these options?
I guess it makes sense to have options for arguments that are almost always used, but for the rest `extraArguments` should be enough. The example for `extraArguments` could also include some common options.

Reply:
I agree, but I think those are the most important. I don't think we should change that.